Line clipping panics when linestring has 0 points

I am working with a Geofabrik country-level (Italy) extract. I have no problems running master which is at 43fa8d9 from Dec 22. Over the weekend I made a build from the v0.6.0 branch so that I could start using environment variables and now when I do the simplest things I get these panic messages
```
2018/02/05 12:20:46 server.go:2848: http: panic serving [::1]:50538: runtime error: index out of range
```
after seeing errors like these. I am not certain they always correspond to queries having 0 results.
```
2018/02/05 12:20:43 map.go:180: Error Getting MVTLayer for tile Z: 10, X: 545, Y: 379: error running layer (pylon_areas) SQL (SELECT ST_AsBinary(geometry) AS geometry, osm_id, type, name, category, water_level FROM osm_pylon_areas WHERE geometry && ST_MakeEnvelope(1.290868533500365e+06,5.205667373609009e+06,1.3312272844293201e+06,5.165308622680054e+06,3857)): read tcp [::1]:50749->[::1]:5432: read: connection reset by peer
```
Please let me know what I can do to help debug this.

--- a/atlas/map.go
+++ b/atlas/map.go
@@ -177,7 +177,7 @@ func (m Map) Encode(ctx context.Context, tile *slippy.Tile) ([]byte, error) {
 					z, x, y := tile.ZXY()
 					// TODO (arolek): should we return an error to the response or just log the error?
 					// we can't just write to the response as the waitgroup is going to write to the response as well
-					log.Printf("Error Getting MVTLayer for tile Z: %v, X: %v, Y: %v: %v", z, x, y, err)
+					log.Printf("err fetching tile (z: %v, x: %v, y: %v) features: %v", z, x, y, err)
 				}
 				return
 			}

--- a/maths/clip/clip.go
+++ b/maths/clip/clip.go
@@ -10,6 +10,9 @@ import (
 
 func LineString(linestr tegola.LineString, extent *points.Extent) (ls []basic.Line, err error) {
 	line := lines.FromTLineString(linestr)
+	if len(line) == 0 {
+		return ls, nil
+	}
 
 	var cpts [][2]float64
 	lptIsIn := extent.Contains(line[0])

--- a/maths/clip/linestring_test.go
+++ b/maths/clip/linestring_test.go
@@ -189,6 +189,10 @@ func TestLineString(t *testing.T) {
 				basic.NewLine(144.397830, 4096, 0, 3901.712895),
 			},
 		},
+		"empty line": tcase{
+			extent:  &testExtents[11],
+			linestr: basic.Line{},
+		},
 	}
 	for name, tc := range tests {
 		tc := tc

